apps/aechannel/src/aesc_fsm.erl

```diff
@@ -2595,10 +2716,25 @@ has_my_signature(Me, SignedTx) ->
             ok =:= aetx_sign:verify_one_pubkey(Me, SignedTx, CurrHeight)
     end.
 
-start_min_depth_watcher(Type, SignedTx, Updates,
+
+maybe_restart_watcher(#data{ watcher = undefined
+                           , on_chain_id = ChId } = D) ->
+    {ok, Pid} = aesc_chain_watcher:restart_watcher(ChId, ?MODULE),
+    D#data{ watcher = Pid }.
+
+start_min_depth_watcher(Type, SignedTx, Updates, D) ->
+    try start_min_depth_watcher_(Type, SignedTx, Updates, D)
+    catch
+        error:E ->
+            T = erlang:get_stacktrace(),
+            lager:debug("CAUGHT E=~p / T = ~p", [E, pr_stacktrace(T)]),
+            error({caught,E})
+    end.
+start_min_depth_watcher_(Type, SignedTx, Updates,
                         #data{ watcher = Watcher0
                              , op = Op
-                             , opts = #{minimum_depth := MinDepth}} = D) ->
+                             , opts = Opts } = D) ->
+    MinDepth = maps:get(minimum_depth, Opts, ?MINIMUM_DEPTH),
```

Comment:
This may seem odd... The purpose for this was to avoid a `function_clause` error in one of the `aest_db_SUITE` system test runs.

Reply:
I will change that later too. So not a problem.